src/app/common/app.controller.js

```diff
@@ -69,12 +69,41 @@
       if ($window.localStorage.getItem('currentUser')) {
         $scope.setCurrentUser(JSON.parse($window.localStorage.getItem('currentUser')));
       }
+      $scope.initLanguage();
     };
 
     $scope.activeNav = function (path) {
       return $location.path() === path;
     };
 
+    $scope.initLanguage = function () {
+      if (!$scope.currentUser) {
+        return;
+      }
+
+      var locale = $scope.currentUser.locale;
+      var lang = gettextCatalog.getCurrentLanguage();
+
+      if (lang !== locale) {
+        gettextCatalog.setCurrentLanguage(locale);
+        $scope.language = locale;
+      }
+    }
+
+    $scope.changeLanguage = function (lang) {
+      gettextCatalog.setCurrentLanguage(lang);
+      $scope.currentUser.locale = lang;
+      $scope.language = lang;
+      User.update($scope.currentUser, function (user) {
```

Comment:
I wouldn't update the user's locale when the user just changes the locale in the dropdown menu in the header. Do this only from the preferences page.